### ISSUE TITLE
CookieManager simple example not working

### ISSUE THREAD

**Issue by nriccar** [react-native-cookies/cookies#59]
Hello, I'm trying to set a cookie 🍪 like in the example documentation and I'm getting this error:

![WhatsApp Image 2020-06-18 at 18 05 06](https://user-images.githubusercontent.com/49196314/85044894-8bf84f00-b18e-11ea-95a8-c9f8239d840a.jpeg)

My code: 
```
CookieManager.set('http://example.com', {
              name: 'myCookie',
              value: 'myValue',
              domain: 'some domain',
              path: '/',
              version: '1',
              expires: '2015-05-30T12:30:00.00-05:00'
            }).then(done => {
              console.log('CookieManager.set =>', done)
            })
```

**Comment by tiborazo** [comment]
Same here. Looks like the name & value are missing the double quotes in v3 - the .set is not working on iOS

**Comment by nriccar** [comment]
Nope, I already have @react-native-community/cookies. Tried to unlink it, uninstall it, and install @react-native-community/cookies@^3.0.0, and I'm having the same error.
Also, the double quotes are the same as the simple ones. If I write double Visual Code converts them to simple.

**Comment by nriccar** [comment]
Anyone?

Thanks

**Comment by Zo2m4bie** [comment]
Same here. Are there any update on this problem?

**Comment by focux** [comment]
I'm having this same issue. For the record, I'm on version 3.

**Comment by Zo2m4bie** [comment]
Ok, during preparing fix, I found a problem. For me it was url without schema. I had www.test.com but it requires http://www.test.com. 
Also this issue can happen if you have random domain as in example:
CookieManager.set('http://example.com', {
              name: 'myCookie',
              value: 'myValue',
              domain: 'some domain',
              path: '/',
              version: '1',
              expires: '2015-05-30T12:30:00.00-05:00',
            }).then((done) => {
              console.log('CookieManager.set =>', done);
            });
Should be changed to
CookieManager.set('http://example.com', {
              name: 'myCookie',
              value: 'myValue',
              domain: 'example.com',
              path: '/',
              version: '1',
              expires: '2015-05-30T12:30:00.00-05:00',
            }).then((done) => {
              console.log('CookieManager.set =>', done);
            });

**Comment by safaiyeh** [comment]
Hey all! Apologies for the delay, I'll be investigating this issue tonight. I believe I know why the issue is there. Will update sometime today

**Comment by nriccar** [comment]
@safaiyeh Thank you for reviewing this. Have you had the chance to push its fix?
thanks!

**Comment by safaiyeh** [comment]
I have a fix, testing on a couple production apps before I make a PR here.

**Comment by TommyLeong** [comment]
@safaiyeh Meaning to say this is a confirmed bug right? Looking forward for the PR. Thanks man!

**Comment by safaiyeh** [comment]
Hi all this ended up being bad documentation.

```
CookieManager.set('http://example.suba.com', {
  name: 'myCookie',
  value: 'myValue',
  domain: 'suba',
  path: '/',
  version: '1',
  expires: '2015-05-30T12:30:00.00-05:00'
}).then((done) => {
  console.log('CookieManager.set =>', done);
});
```

The `domain` attribute needs to be a subdomain of the URL. In the example above, the only valid domains are `example` or `suba`. Action item from this issue is to update the docs.

Let me know if this is helpful. The fix I mentioned changed the intended behavior.

**Comment by safaiyeh** [comment]
The error also should not be that random. A better exception like "domain does not exist in URL" would be more clear.

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to:
- [`@preeternal/react-native-cookie-manager`](https://github.com/Preeternal/react-native-cookie-manager)
- [`react-native-nitro-cookies`](https://github.com/l2hyunwoo/react-native-nitro-cookies)

Thank you for your contributions! ❤️
